### PR TITLE
codespell fix typos in the main cvmfs/ codebase, add codespell workflow to catch any new typos

### PR DESCRIPTION
**Pull request by yarikoptic**
- [x] workflow was moved to a separate PR
- [x] ~~one typo fix is "functional": `quarantaine/` folder is being renamed to `quarantine/`.  May be should be reverted?~~ reverted, thus no more pertinent.
- there are some typos still left in test cases happen someone wants to practice codespell'-ing ;)

### CODE REVIEW

**Comment by yarikoptic** [comment]
the longer PR stays unattended, the more conflicts it would encounter, the more effort for typos-free-glorious-cvmfs would need to be invested.  Please advise on either I should bother addressing the conflicts.

File: cvmfs/publish/cmd_zpipe.cc [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 #include "duplex_zlib.h"
 
-#if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(__CYGWIN__)
```
Comment (jblomer) [inline cvmfs/publish/cmd_zpipe.cc]: Needs to be reverted

File: cvmfs/swissknife_zpipe.cc [hunk 2]
```diff
@@ -26,7 +26,7 @@
 #include "duplex_zlib.h"
 #include "swissknife_zpipe.h"
 
-#if defined(MSDOS) || defined(OS2) || defined(WIN32) || defined(__CYGWIN__)
```
Comment (jblomer) [inline cvmfs/swissknife_zpipe.cc]: Needs to be reverted

**Review by jblomer — CHANGES_REQUESTED** [review]
Many thanks for the contribution!

The rename of the folder indeed needs to be reverted.

Also, can the workflow please be taken out as a separate PR.

File: cvmfs/swissknife_zpipe.cc (hunk 2 above)
Comment (yarikoptic) [reply cvmfs/swissknife_zpipe.cc]: sorry about that, good job spotting! I will just exclude `msdos` entirely.

**Comment by yarikoptic** [comment]
done, ready for re-review

**Review by jblomer — APPROVED** [review]

**Comment by jblomer** [comment]
@cernvm-bot cpplint

**Comment by jblomer** [comment]
@cernvm-bot tidy

**Comment by jblomer** [comment]
@cernvm-bot unittest

**Comment by jblomer** [comment]
@cernvm-bot cloudtest

**Comment by jblomer** [comment]
@yarikoptic in principle looks good to me, after addressing the linter errors this is good to go.

**Comment by yarikoptic** [comment]
FWIW: lint issues were addressed (I even had `python2` locally to ensure that ;))

**Comment by jblomer** [comment]
@cernvm-bot unittest

**Pull request merged**